--- SConstruct.orig	2021-01-18 19:29:07.000000000 -0800
+++ SConstruct	2021-01-18 21:29:48.000000000 -0800
@@ -1211,9 +1211,14 @@
     # is basically taken from parts of the code for TryBuild and TryRun.
 
     # First make the file name using the same counter as TryBuild uses:
-    f = "conftest_" + str(SCons.SConf._ac_build_counter)
-    SCons.SConf._ac_build_counter = SCons.SConf._ac_build_counter + 1
-
+    if int(SCons.__version__.split('.')[0]) >= 4:
+        testSig = SCons.Util.MD5signature(pname)
+        f = "conftest_" + str(SCons.SConf._ac_build_counter[testSig])
+        SCons.SConf._ac_build_counter[testSig] += 1
+    else:
+        f = "conftest_" + str(SCons.SConf._ac_build_counter)
+        SCons.SConf._ac_build_counter += 1
+    
     config.sconf.pspawn = config.sconf.env['PSPAWN']
     save_spawn = config.sconf.env['SPAWN']
     config.sconf.env['SPAWN'] = config.sconf.pspawn_wrapper
