--- setup.py.orig	2022-01-08 17:39:43.000000000 -0800
+++ setup.py	2022-01-08 17:37:05.000000000 -0800
@@ -74,7 +74,7 @@
 shared_data = all_files_from('share')
 
 copt =  {
-    'gcc' : ['-O2','-msse2','-std=c++11','-fvisibility=hidden','-fopenmp'],
+    'gcc' : ['-O2','-std=c++11','-fvisibility=hidden','-fopenmp'],
     'icc' : ['-O2','-msse2','-vec-report0','-std=c++11','-openmp'],
     'clang' : ['-O2','-msse2','-std=c++11',
                '-Wno-shorten-64-to-32','-fvisibility=hidden','-stdlib=libc++'],
