--- a/setup.py
+++ b/setup.py
@@ -1023,7 +1023,7 @@
                     lflags.append('-Wl,-rpath,' + fftw_libpath)
                 # This should work, but for some reason it doesn't.
                 # So I'm leaving it here, but we'll fix it below with install_name_tool anyway.
-                lflags.append('-install_name @rpath/%s'%full_lib_name)
+                lflags.append('-Wl,-install_name,@rpath/%s'%full_lib_name)
 
             output_dir = os.path.join('build','shared_clib')
             self.compiler.link(CCompiler.SHARED_OBJECT, expected_objects, full_lib_name,
